search-service/src/test/kotlin/com/egm/stellio/search/service/AttributeServiceTest.kt

```diff
@@ -0,0 +1,247 @@
+import com.egm.stellio.shared.util.JsonLdUtils.NGSILD_LOCATION_TERM
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.data.r2dbc.core.R2dbcEntityTemplate
+import org.springframework.r2dbc.core.DatabaseClient
+import org.springframework.test.context.ActiveProfiles
+import java.time.Instant
+import java.time.ZoneOffset
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@SpringBootTest
+@ActiveProfiles("test")
+class AttributeServiceTest : WithTimescaleContainer, WithKafkaContainer {
+
+    @Autowired
+    private lateinit var attributeService: AttributeService
+
+    @Autowired
+    private lateinit var databaseClient: DatabaseClient
+
+    @Autowired
+    private lateinit var r2dbcEntityTemplate: R2dbcEntityTemplate
+
+    private val now = Instant.now().atZone(ZoneOffset.UTC)
+
+    private val entityPayload1 = newEntityPayload(
+        "urn:ngsi-ld:BeeHive:TESTA",
+        listOf(BEEHIVE_TYPE, SENSOR_TYPE),
+        DEVICE_COMPACT_TYPE
+    )
+    private val entityPayload2 = newEntityPayload("urn:ngsi-ld:Sensor:TESTB", listOf(SENSOR_TYPE), DEVICE_COMPACT_TYPE)
+    private val entityPayload3 = newEntityPayload("urn:ngsi-ld:Apiary:TESTC", listOf(APIARY_TYPE), DEVICE_COMPACT_TYPE)
+    private val temporalEntityAttribute1 = newTemporalEntityAttribute(
+        "urn:ngsi-ld:BeeHive:TESTA",
+        INCOMING_PROPERTY,
+        TemporalEntityAttribute.AttributeType.Property,
+        TemporalEntityAttribute.AttributeValueType.NUMBER
+    )
+    private val temporalEntityAttribute2 = newTemporalEntityAttribute(
+        "urn:ngsi-ld:BeeHive:TESTA",
+        MANAGED_BY_RELATIONSHIP,
+        TemporalEntityAttribute.AttributeType.Relationship,
+        TemporalEntityAttribute.AttributeValueType.STRING
+    )
+    private val temporalEntityAttribute3 = newTemporalEntityAttribute(
+        "urn:ngsi-ld:Apiary:TESTC",
+        NGSILD_LOCATION_PROPERTY,
+        TemporalEntityAttribute.AttributeType.GeoProperty,
+        TemporalEntityAttribute.AttributeValueType.GEOMETRY
+    )
+    private val temporalEntityAttribute4 = newTemporalEntityAttribute(
+        "urn:ngsi-ld:Sensor:TESTB",
+        OUTGOING_PROPERTY,
+        TemporalEntityAttribute.AttributeType.Property,
+        TemporalEntityAttribute.AttributeValueType.GEOMETRY
+    )
+
+    @AfterEach
+    fun clearPreviousTemporalEntityAttributesAndObservations() {
+        r2dbcEntityTemplate.delete(EntityPayload::class.java)
+            .all()
+            .block()
+        r2dbcEntityTemplate.delete(AttributeInstance::class.java)
+            .all()
+            .block()
```

Comment:
I don't think you are creating things in attribute_instance table?

Reply:
Yes I remove it